src/esm_runscripts/namelists.py

```diff
@@ -126,46 +126,77 @@ def nmls_remove(mconfig):
 
         namelist_changes = mconfig.get("namelist_changes", {})
         namelist_removes = []
+        error_message = (
+            "There is a syntax error, probably in your runscript, regarding namelist "
+            "changes (e.g. in a 'add_namelist_changes' block). "
+            "It seems that either 'namelist_file' or "
+            "'namelist_group' or 'both' are missing.\n"
```

Comment:
I've looked at the code and it's actually possible to give a hint to what is exactly wrong. For instance, because you have the variable `namelist`, `changes` and `change_chapter` (only on the second if), you can actually point to there is something wrong with that. For example
```
This is wrong:
    [add_]namelist_changes:
        fort.4:
            LHVOLCA: false
The correct syntax would be:
     [add_]namelist_changes:
            <namelist_file>:
                    <namelist_group>:
                            <variable>: <value>
```